src/Microsoft.AspNetCore.Mvc/DesignTimeMvcServiceCollectionProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    internal class DesignTimeMvcServiceCollectionProvider
```

Comment:
Not sure how I feel about this being internal. Aka, we're using it as if it were public in the RazorTooling package but it's actually hidden. We don't do that elsewhere with the .Internal namespaces so I wouldn't envision us doing it here.